korman/operators/op_mesh.py

```diff
@@ -18,12 +18,133 @@
 import math
 import mathutils
 
+from ..exporter import utils
+
 class PlasmaMeshOperator:
     @classmethod
     def poll(cls, context):
         return context.scene.render.engine == "PLASMA_GAME"
 
 
+class PlasmaAddFlareOperator(PlasmaMeshOperator, bpy.types.Operator):
+    bl_idname = "mesh.plasma_flare_add"
+    bl_label = "Add Lamp Flare"
+    bl_category = "Plasma"
+    bl_description = "Adds a new Plasma Lamp Flare"
+    bl_options = {"REGISTER", "UNDO"}
+
+    # Allows user to specify their own name stem
+    flare_name = bpy.props.StringProperty(name="Name",
+                                         description="Flare name stem",
+                                         default="Flare",
+                                         options=set())
+    flare_distance = bpy.props.FloatProperty(name="Distance",
+                                             description="Flare's distance from the illuminating object",
+                                             min=0.1, max=2.0, step=10, precision=1, default=1.0,
+                                             options=set())
+
+    @classmethod
+    def poll(cls, context):
+        return super().poll(context) and context.mode == "OBJECT"
+
+    def draw(self, context):
+        layout = self.layout
+        box = layout.box()
+        box.label("Flare Name:")
+        row = box.row()
+        row.alert = not self.flare_name
+        row.prop(self, "flare_name", text="")
+
+        box = layout.box()
+        box.label("Geometry:")
+        row = box.row()
+        row.prop(self, "flare_distance")
+
+    def execute(self, context):
+        if context.space_data.local_view:
+            bpy.ops.view3d.localview()
+        self.create_flare_objects()
+        return {"FINISHED"}
+
+    def create_flare_objects(self):
+        bpyscene = bpy.context.scene
+        cursor_shift = mathutils.Matrix.Translation(bpy.context.scene.cursor_location)
+
+        for obj in bpy.data.objects:
+            obj.select = False
+
+        # Create Empty, rotated to horizontal
+        flare_root = bpy.data.objects.new("{}".format(self.name_stem), None)
+        flare_root.empty_draw_type = "SINGLE_ARROW"
+        flare_root.matrix_world = cursor_shift
+        flare_root.rotation_euler[0] = math.pi / 2
```

Comment:
Why not `math.radians(90)`? Requires less thinking on my part :)

Reply:
I guess I naturally think in terms of rotations here instead of degrees, since that's the goal of this, so `pi / 2` made the most sense.  I'll change it though so it'll be consistent.